src/Strings.f90

```diff
@@ -1270,35 +1366,31 @@ FUNCTION isFloat(this) RESULT(bool)
     LOGICAL(SBK) :: bool
     INTEGER(SIK) :: stt,dec_pos,exp_pos,pos_neg_exp
 
-    ! Look for '+' and '-'
-    stt = MAX(INDEX(this%s,'+'),INDEX(this%s,'-'))
-    ! Find the decimal if there is one
-    dec_pos = INDEX(this%s,'.')
-    ! Look for Currently acceptable exponents
-    exp_pos = MAX(INDEX(this%s,'e'),INDEX(this%s,'E'),INDEX(this%s,'d'))
-    pos_neg_exp = MAX(INDEX(this%s,'-'),INDEX(this%s,'+'))
-
+    bool = .FALSE.
     ! Enforce that a decimal is required in order to be a float
+    dec_pos = INDEX(this%s,'.')
     IF(dec_pos > 0) THEN
-      ! Check potential number leading up to decimal
-      bool = isNumeric(this%s(stt+1:dec_pos-1))
-      ! There is a possible exponent
-      IF(exp_pos > 0) THEN
-        ! Check number between decimal and exponent
-        bool = bool .AND. isNumeric(this%s(dec_pos+1:exp_pos-1))
-        ! Account for any sign on the exponent
-        IF(pos_neg_exp > 0) THEN
-          bool = bool .AND. isNumeric(this%s(pos_neg_exp+1:LEN(this%s)))
+      ! Look for leading '+' and '-'
+      stt = MAX(INDEX(this%s(1:dec_pos),'+'),INDEX(this%s(1:dec_pos),'-'))
+      ! Check that the leading characters are numeric
+      IF(isNumeric(this%s(stt+1:dec_pos-1))) THEN
+        ! Look for Currently acceptable exponents
+        exp_pos = MAX(INDEX(this%s,'e'),INDEX(this%s,'E'),INDEX(this%s,'d'))
+        IF(exp_pos > 0) THEN
+          ! Account for sign on exponent
+          pos_neg_exp = exp_pos + MAX(INDEX(this%s(exp_pos+1:LEN(this%s)),'-'),INDEX(this%s(exp_pos+1:LEN(this%s)),'+'))
+          IF(pos_neg_exp > 0) THEN
+            bool = isNumeric(this%s(dec_pos+1:exp_pos-1)) .AND. &
+                isNumeric(this%s(pos_neg_exp+1:LEN(this%s)))
+          ELSE
+            bool = isNumeric(this%s(dec_pos+1:exp_pos-1)) .AND. &
+                isNumeric(this%s(exp_pos+1:LEN(this%s)))
+          ENDIF
```

Comment:
The idea would be testing user inputs prior to "_casting_" to reals. Which leads into the checking of numbers after the exponent, since they would be input by a user they could feasibly do something dumb like 3.5eXXX. Most of this is just an effort to avoid IOerrors and provide meaningful feedback.

Reply:
Ok, sounds good.  Hopefully these routines are only called during input processing so they don't have a discernible impact on runtime, which was my thinking with checking only between the decimal and exponent.